akd/src/append_only_zks.rs

```diff
@@ -152,15 +153,62 @@ impl Azks {
         storage: &StorageManager<S>,
         insertion_set: &[Node],
     ) -> Result<u64, AkdError> {
-        let prefixes_set = crate::utils::build_prefixes_set(
-            insertion_set
-                .iter()
-                .map(|n| n.label)
-                .collect::<Vec<NodeLabel>>()
-                .as_ref(),
-        );
+        let insertion_labels = insertion_set
+            .iter()
+            .map(|n| n.label)
+            .collect::<Vec<NodeLabel>>();
+
+        self.bfs_preload_nodes_sorted::<S>(storage, &insertion_labels)
```

Comment:
q: Hmm I feel like there's got to be a way to not have to collect this set and then reference it again no?

Reply:
True, we could avoid this, but I think it comes at some additional complexity to the code. I benchmarked this operation and for inserting 100k nodes, it only takes 6 milliseconds to do this collect. So maybe not worth the optimization for now? Unless you see a simple way to address it.